office-js/index.d.ts

```diff
@@ -1419,334 +1425,174 @@ declare namespace Office {
-            getEntities(): Office.Entities;
+            Declined
+        }
+        export enum RecipientType {
             /**
-             * Gets an array of entities of the specified entity type found in an appointment
-             * @param entityType One of the EntityType enumeration values
+             * Specifies that the recipient is a distribution list containing a list of email addresses
              */
-            getEntitiesByType(entityType: Office.MailboxEnums.EntityType): Office.Entities;
+            DistributionList,
             /**
-             * Returns well-known entities that pass the named filter defined in the manifest XML file
-             * @param name  A TableData object with the headers and rows
+             * Specifies that the recipient is an SMTP email address that is on the Exchange server
              */
-            getFilteredEntitiesByName(name: string): Office.Entities;
+            User,
             /**
-             * Returns string values in the currently selected appointment object that match the regular expressions defined in the manifest XML file
+             * Specifies that the recipient is an SMTP email address that is not on the Exchange server
              */
-            getRegExMatches(): string[];
+            ExternalUser,
             /**
-             * Returns string values that match the named regular expression defined in the manifest XML file
+             * Specifies that the recipient is not one of the other recipient types
              */
-            getRegExMatchesByName(name: string): string[];
+            Other
+        }
+        export enum RestVersion {
+            v1_0,
+            v2_0,
+            Beta
         }
     }
     export module cast {
         export module item {
-            function toAppointmentCompose(item: Office.Item): Office.Types.AppointmentCompose;
-            function toAppointmentRead(item: Office.Item): Office.Types.AppointmentRead;
+            function toAppointmentCompose(item: Office.Item): Office.AppointmentCompose;
+            function toAppointmentRead(item: Office.Item): Office.AppointmentRead;
             function toAppointment(item: Office.Item): Office.Appointment;
-            function toMessageCompose(item: Office.Item): Office.Types.MessageCompose;
-            function toMessageRead(item: Office.Item): Office.Types.MessageRead;
+            function toMessageCompose(item: Office.Item): Office.MessageCompose;
+            function toMessageRead(item: Office.Item): Office.MessageRead;
             function toMessage(item: Office.Item): Office.Message;
-            function toItemCompose(item: Office.Item): Office.Types.ItemCompose;
-            function toItemRead(item: Office.Item): Office.Types.ItemRead;
+            function toItemCompose(item: Office.Item): Office.ItemCompose;
+            function toItemRead(item: Office.Item): Office.ItemRead;
         }
     }
+    export interface asyncContextOptions {
```

Comment:
I think that this would be the expected behavior. While, technically, you can put anything in the options parameter, and the code will work, the only value we care about is asyncContext, any other values in the options parameter are likely to be mistakes made by the user.